articles/quickstart/webapp/nodejs-ea/01-login.md

```diff
@@ -0,0 +1,151 @@
+---
+title: Login
+description: This tutorial demonstrates how to add user login, logout, and profile to a Node.js Express application.
+budicon: 448
+topics:
+  - quickstarts
+  - webapp
+  - login
+  - user profile
+  - logout
+  - nodejs
+contentType: tutorial
+useCase: quickstart
+github:
+  path: 01-Login
+---
+
+<!-- markdownlint-disable MD002 -->
+## Getting Started
+In this quickstart you will learn how to implement authentication in an Express application using the OIDC Compliant library [Express OIDC](https://github.com/auth0/express-openid-connect) authored and maintained by Auth0.
+
+::: warning
+Express OIDC is still in an [early access stage](/).
+:::
+
+### Install Dependencies
+
+```sh
+# Using NPM
+npm install express-openid-connect express-session
+
+# Using Yarn
+yarn add express-openid-connect express-session
+```
+
+## Configure Auth0
+### Configure Callback URLs
+
+A callback URL is a URL in your application where Auth0 redirects the user after they have authenticated. 
+
+The callback URL for your app must be whitelisted in the **Allowed Callback URLs** field in your [Application Settings](${manage_url}/#/applications/${account.clientId}/settings). If this field is not set, users will be unable to log in to the application and will get an error.
+
+::: note
+If you are testing your application locally, you should set the **Allowed Callback URL** to `http://localhost:3000/callback`
+:::
+
+### Configure Logout URLs
+
+A logout URL is a URL in your application that Auth0 can return to after the user has been logged out of the authorization server. This is specified in the `returnTo` query parameter.
+
+The logout URL for your app must be whitelisted in the **Allowed Logout URLs** field in your [Application Settings](${manage_url}/#/applications/${account.clientId}/settings). If this field is not set, users will be unable to log out from the application and will get an error.
+
+::: note
+If you are testing your application locally, the logout URL you need to whitelist in the **Allowed Logout URLs** field is `http://localhost:3000`.
+:::
+
+## Integrate Auth0
+### Auth0 Application Keys
+
+You will need to retrieve some details for your application to communicate with Auth0. You can get these details from the [Application Settings](${manage_url}/#/applications/${account.clientId}/settings) section in the Auth0 dashboard.
+
+You need the following information:
+
+* **Domain**
+* **Client ID**
+* **Client Secret**
+
+### Express Middleware
+There are three middlewares required to integrate Auth0 in your application.
+
+- `express.urlencoded` - parses urlencoded payloads sent back from the Auth0 server.
+- `express-session` - simple session middleware for Express (other session middleware can be used as a replacement).
```

Comment:
I don't think this is technically a middleware. Also, we probably just want to call out that there needs to be session handling of some kind. They can use `express-session` (stores in memory ... there was a conversation at some point internally about how this might not be ideal) or `cookie-session` or another session hander. We'll need to pick one for this QS (and be confident we're choosing the right one) but I think we should make it clear that either will work.

Reply:
Yes I noticed `cookie-session` was used in the internal doc, but used the more widely used library `express-session`.  We should pick one and stick with it.  Don't need to worry the user with choice here.

Reply:
I agree ... but I want to make sure we're not *mandating* `express-session`. Cookie session works fine as well. Point being: there needs to be a session handler and the one we choose here is A or B. Just like env management in that sense.

Reply:
> but I think we should make it clear that either will work.

the wording `other session middleware can be used as a replacement` is possibly poor, but this was the intent

Reply:
@joshcanhelp lets revisit this after I chunk out this section